### DISCUSSION TITLE
Kura Wires - j2mod.modbus - Times out on all writes even when successful

### DISCUSSION THREAD

**Discussion by rwskinner** [eclipse-kura/kura#5329] [Q&A]
I'm using the latest modbus driver V2.04 (July 8th, 2024) in my wire graph.  I'm running in TCP mode.  It works fine reading, but when writing, it returns a timeout, even though the value is written to the sever fine.  I increased the timeout of the driver all the way to 20 seconds and it continues to occur.    I tried it writing multiple registers to a float, and also single registers to a single integer register.

Error:
Reason: com.ghgande.j2mod.modbus.ModbusIOException: Executing transaction 53 B6 00 00 00 0B C8 10 03 47 00 02 04 3F 66 66 66 failed (tried 1 times) Socket timeout reading response - Read timed out

If I use Modbus Poll to write then it writes the value fine, shows success, and has the proper value under communications.
FC16 Mode
Tx:001948-07 79 00 00 00 0B C8 10 03 47 00 02 04 3F 66 66 66
Rx:001949-07 79 00 00 00 0B C8 10 03 47 00 02

Writing to a single register (integer).  Fails with the same timeout.
Reason: com.ghgande.j2mod.modbus.ModbusIOException: Executing transaction 53 C4 00 00 00 09 C8 10 02 BB 00 01 02 00 01 failed (tried 1 times) Socket timeout reading response - Read timed out

Modbus Poll writes to it fine in FC6 Mode
Tx:001954-07 91 00 00 00 06 C8 06 02 BB 00 01
Rx:001955-07 91 00 00 00 06 C8 06 02 BB 00 01

In FC16 Mode:
Tx:001956-07 9D 00 00 00 09 C8 10 02 BB 00 01 02 00 01
Rx:001957-07 9D 00 00 00 09 C8 10 02 BB 00 01

Has anyone else reproduced this?

**Comment by MMaiero** [comment]
Hello,
the driver you are referring to is provided and distributed by a third party (Eurotech) and not by this open source project.
You should refer to the Eurotech support for getting assistance.

Best regards
